Task nxdoc 2696 2023 to 2025 upgrade (#2381)
nuxeo/doc.nuxeo.com-content: src/nxdoc/nuxeo-server/upgrading-the-nuxeo-platform/how-to-upgrade-from-lts-2023-to-lts-2025/how-to-upgrade-to-jakarta-ee-10.md

  </dependency>
  <dependency>
    <groupId>jakarta.persistence</groupId>
    <artifactId>jakarta.persistence-api</artifactId>
  </dependency>
  <dependency>
    <groupId>jakarta.servlet</groupId>
    <artifactId>jakarta.servlet-api</artifactId>
  </dependency>
  <dependency>
    <groupId>jakarta.servlet.jsp</groupId>
    <artifactId>jakarta.servlet.jsp-api</artifactId>
  </dependency>
  <dependency>
    <groupId>jakarta.servlet.jsp.jstl</groupId>
    <artifactId>jakarta.servlet.jsp.jstl-api</artifactId>
  </dependency>
  <dependency>
    <groupId>jakarta.transaction</groupId>
    <artifactId>jakarta.transaction-api</artifactId>
  </dependency>
  <dependency>
    <groupId>jakarta.validation</groupId>
    <artifactId>jakarta.validation-api</artifactId>
  </dependency>
  <dependency>
    <groupId>jakarta.ws.rs</groupId>
    <artifactId>jakarta.ws.rs-api</artifactId>
  </dependency>
  <dependency>
    <groupId>jakarta.xml.bind</groupId>
    <artifactId>jakarta.xml.bind-api</artifactId>
  </dependency>
  <dependency>
    <groupId>jakarta.xml.ws</groupId>
    <artifactId>jakarta.xml.ws-api</artifactId>
  </dependency>
  ..
</dependencies>
```

### Implementations

> The implementations have changed too, some have just been renamed to jakarta, others have been changed to use other vendor.

Reviewer: ```suggestion

The implementations have also changed. Some have just been renamed Jakarta, and others have been changed to use another vendor.

> Generallyn a project doesn't state the Implementations it is using, it might just depend on the Specifications. If your 

Reviewer: ```suggestion
Usually, a project doesn't state the Implementations it is using, it may depend on the Specifications. If your 
project is using implementations, you will need to update them, see below the one used by Nuxeo Server:
```

project is using implementations, you will need to update them, see below the one used by Nuxeo Server:

```xml
<dependencies>
  ..
  <dependency>
    <groupId>com.sun.activation</groupId>
    <artifactId>jakarta.activation</artifactId>
  </dependency>
  <dependency>
    <groupId>org.apache.tomcat</groupId>
    <artifactId>tomcat-jasper-el</artifactId>
  </dependency>
  <dependency>
    <groupId>org.eclipse.angus</groupId>
    <artifactId>jakarta.mail</artifactId>
  </dependency>
  <dependency>
    <groupId>org.apache.geronimo.components</groupId>
    <artifactId>geronimo-transaction</artifactId>
  </dependency>
  <dependency>
    <groupId>org.apache.tomcat</groupId>
    <artifactId>tomcat-*</artifactId>
  </dependency>
  <dependency>
    <groupId>org.glassfish.jersey.core</groupId>
    <artifactId>jersey-*</artifactId>
  </dependency>
  <dependency>
    <groupId>com.sun.xml.bind</groupId>
    <artifactId>jaxb-impl</artifactId>
  </dependency>
  ..
</dependencies>
```

### Libraries

Several libraries leverage Java Specifications, with this upgrade to Jakarta EE 10, they might be updated too.

> Updating the library version is enough in most cases, so we encourage you to update all the dependencies you may depend on
> and that are not given by Nuxeo Parent pom.

Reviewer: ```suggestion
In most cases, updating the library version is enough, so we encourage you to update all the dependencies you may depend on and that are not provided by Nuxeo Parent pom.
```

> In some cases, you might find an appropriate alternative, this was the case for `jackson-jaxrs-json-provider` that 

Reviewer: ```suggestion
In other cases, you might find an appropriate alternative, this was the case for `jackson-jaxrs-json-provider` that 
```

---
title: How to upgrade to Jakarta EE 10

> description: Instruction to upgrade your Nuxeo Project to LTS 2025 - Jakarta EE part

Reviewer 2: ```suggestion
description: Instructions to upgrade your Nuxeo Project to LTS 2025 - Jakarta EE part
```
